src/components/mixins/DownloadMixin.vue

```diff
@@ -42,14 +47,59 @@ export default class DownloadMixin extends Vue {
         return this.$store.getters['profile/activeProfile'];
     }
 
-    async downloadCompletedCallback(downloadedMods: ThunderstoreCombo[]): Promise<void> {
+    async downloadCompletedCallback(downloadedMods: ThunderstoreCombo[], assignId: number): Promise<void> {
         try {
-            await installModsAndResolveConflicts(downloadedMods, this.profile.asImmutableProfile(), this.$store);
+            await this.installModsAndResolveConflicts(downloadedMods, this.profile.asImmutableProfile(), this.$store, assignId);
         } catch (e) {
+            this.$store.commit('download/updateDownload', {assignId, failed: true});
             this.$store.commit('error/handleError', R2Error.fromThrownValue(e));
         }
     }
 
+    downloadProgressCallback(assignId: number, downloadProgress: number, modName: string, status: number, err: R2Error | null) {
+        try {
+            if (status === StatusEnum.FAILURE) {
+                this.setIsModProgressModalOpen(false);
+                this.$store.commit('download/updateDownload', {assignId, failed: true});
+                if (err !== null) {
+                    DownloadMixin.addSolutionsToError(err);
+                    throw err;
+                }
+            } else if (status === StatusEnum.PENDING || status === StatusEnum.SUCCESS) {
```

Comment:
Why does this differ from L83? I think these two implementations should be identical, unless there's a very good reason for them not to be.

~In fact, if they should be identical, I think this should by DRYed up by calling `DownloadMixin.downloadProgressCallback` in the try-catch here. (Also, is the try-catch required here? It might be, I can't recall.)~

In fact, I think the place where this is called should call the static method instead. The try-catch seems (please verify) unnecessary since the call site already has error handling. But do check if `|| status === StatusEnum.SUCCESS` should also be applied to the static method too.

Reply:
Now uses the static function for both 👍